### DISCUSSION TITLE
Grizzl-E URL Format??? Maybe something else? Cant connect.

### DISCUSSION THREAD

**Discussion by ericvogel84** [lbbrhzn/ocpp#800] [Q&A]
I've got the integration running and have tried connecting my Grizzl-E about 30 or so times, formatting my URL every which way and I just cant get it to successfully connect - it just seems to time out and not connect to my router. It does connect via the chargelab app setup process just fine. 

Process goes:

- Reset DIP switches
- Connect to charger wifi network
- Enter wifi credentials
- Choose "Other" network
- Enter localip:9000 or homeassistant.local:9000 or ws://localip:9000 or ws://localip:9000/ and all variations of those
- Charger page accepts and attempts to connect
- Charger wifi network becomes available again and it starts over

The esp-arduino device never shows up on my router as having connected, and I'm not getting anything in the HA logs. I have also tried resetting my OCCP port on HA. 

Any help would be apprecaited!

**Comment by BjarneBitscrambler** [comment]
Eric - have you tried adding the Grizzl-E's serial number to the portal address after the trailing forward-slash (/)?  For example, on my charger I enter ```10.0.0.17:9000/GRS-0000000023f8``` for the connection address.  I don't believe there is any check made that the serial number is correct, but it can't be blank.

**Reply by ericvogel84** [reply]
Thanks for your reply! 

I gave this a shot, and still nothing. However, I do think you’re right that the URL needs to have a serial/ID - I just located this “Developer” portion of the Grizzl-E site that talks about it. https://grizzl-e.com/grizzl-e-smart-developers/

I tried a few different things (using different formatting of my serial), including using just the same device name that I used in Home Assistant for the ID/Serial - IE 192.168.7.223:9000/charger 

Bummer- I really thought for a moment this was the quick solution I was looking for!

**Comment by BjarneBitscrambler** [comment]
That's frustrating...  OK, my next two thoughts are: firmware problem; or router incompatibility.  Do you know the firmware version on your Grizzl-E?  Perhaps it's too old, or too new - others including me using the HA integration have been successful with v5.663.  Version 6 was announced in the fall, but as far as I know, no-one has tried it yet. Unfortunately, to find out the firmware version you either have to submit a support ticket with Grizzl-E (click on the Technical Support link at the bottom of the grizzl-e.com webpage, and click through to the Submit A Request form), or have the connection to HA already working (so that option is no good for you).

To check the router isn't the problem, ensure that you aren't using a "Guest" Access Point - the Grizzl-E has to be able to communicate with other devices on the local net (i.e. the HA server). If you connect a laptop to the same AP, are you able to ping the HA server from the laptop using the same local address you gave the Grizzl-E?  Another possible check would be to setup the device running the HA server as it's own AP, then tell the Grizzl-E to connect to that AP (thereby bypassing the router).

**Reply by ericvogel84** [reply]
I'll request the firmware info from Grizzl-E - no idea as of right now. I just bought it back in December, so it's likely to be a pretty recent version, unless it was sitting on a shelf somewhere for a long time. None of the screenshots from the manual seem to directly match what I'm seeing, so it's for sure not one of the really old ones. 

I'm *pretty* confident that the router isnt the problem, as it's the same AP/password that is successful when I use the chargelab app to set it up. I also tried using a 2.4Ghz wifi/ethernet bridge/extender that is on the same subnet as the HA server (and have verified access to the HA server via a laptop connected to it), and it will not connect to that either. 

It would make sense to me that something was up with HA/the integration if it would connect to the AP and then just sit there/ fail to talk to the HA server, but it doesnt seem to ever connect to the AP at all. I'm to the point where I think there must be a bug in the firmware that I'm on... At this point, that's much easier to stomach than the idea that I'm just missing something really simple lol 


Thanks again for all your time responding here! :)

**Comment by BjarneBitscrambler** [comment]
If you feel like poking at the problem some more while waiting for Grizzl-E / Chargelabs to reply, then I can think of a few deeper tests to confirm the connection between the Charger and the HA Server.  

Your confirmation that the Charger communicates OK with Chargelabs when using your WiFi AP does indeed suggest that it isn't a radio signal strength issue, nor a SSID/Password issue. But, we can't still be sure whether or not the Grizzl-E is trying to talk to the HA server.  If you feel ambitious, you can install Wireshark or BurpSuite (these are packet-sniffing tools; I used the latter) on a laptop connected to your local net.  Then configure BurpSuite to forward all traffic coming in on port 9000 of the laptop's IP, to port 9000 of the HA server's IP.  (Alternatively, you could skip this step if you install the packet-sniffer on the same computer that is running HA server) Then tell your Grizzl-E to connect to the IP address of the laptop.  The packet-sniffer should capture and display a continuous stream of websocket traffic between the Grizzl-E and the HA server.  If not, then it should be informative to note whether there is *any* attempt at all by the Grizzl-E to communicate on the local net, and what the error(s) are that prevent it from opening a websocket connection.

Anyway, this might be going into more detail than you are inclined to try...either way, good luck, and let us know how things progress!

**Reply by ericvogel84** [reply]
I just installed Wireshark on my Unraid server, used its IP address w/port 9000 for the charger address, and filtered the display for all TCP and UDP traffic @ port 9000 on the server - no logs appeared after clicking submit on the charger web-portal/setup and waiting a couple minutes until it reset.

It does indeed appear that for whatever reason, my charger is not successful in joining the WiFi network unless I use the ChargeLab app. I've opened a support ticket with them to see what they say, but don't expect them to be particularly helpful. :/

**Reply by drc38** [reply]
Try just filtering on the charger's IP address. You should see messages between it and Chargelab server. It may be that it is using a different port to that specified for some reason eg secure port number.

**Comment by ericvogel84** [comment]
I have to say I’m pretty impressed with the honesty here. Lol

From Grizzl-e support:

“We are not sure why the charger is not establishing communication with the IP address. We suggest defining the server DNS as something other than the IP address.
 
To be honest, this version of Grizzl-E Smart was developed by a different team, and our current developers do not have many insights into how it works. We are releasing a new OCPP firmware version that will hopefully resolve most of these issues.”

**Comment by BjarneBitscrambler** [comment]
I suppose trying different variations of the HA server's address is worth a try. Using a name rather than an IP address would cause a query to the local DNS, which should show up in Wireshark and the DNS server's logs as a name lookup.

Something you said earlier ("...esp-arduino device never shows up on my router as having connected") was puzzling me.  Are you saying that when you point the Grizzl-E at Chargelab's default portal, the Grizzl-E shows up as connected in the router's logs?  But when you point the Grizzl-E at your HA server, the Grizzl-E *doesn't* show up in the router's list?  Weird...   Is there another router you can borrow for a while to see if the same behaviour occurs?

That's interesting about the firmware development history.  I wonder if United Chargers has broken off their association with ChargeLabs?

**Reply by ericvogel84** [reply]
I have continued to try both the IP address and .local address of the HA server, with no change in behavior. I also attempted to use a Wi-Fi extender as an AP (separate SSID from the main router) and the behavior was also the same. 

Yeah I cant seem to figure out what’s going on with ChargeLabs and United Chargers. I had contacted ChargeLab originally and they just shrugged their shoulders. After asking them about the 6.0 firmware, and they said that it wasn’t ready yet - so at least it seems like they are on *some* sort of path over there. Kind of seems like a real mess though.
